Cython/Compiler/Naming.py

```diff
@@ -73,6 +73,24 @@
     'umethod': pyrex_prefix + "umethod_",
 }
 
+
+class Global:
```

Comment:
I was also wondering whether I the `__str__` conversion is slightly too clever... I mean it works beautifully for all the `%s` format strings (short of having python3 and a `__format__` which would have been clearer).

I'm a bit more skeptical of making it a string subclass. Conceptually it feels like it's more a tuple of 2 strings, rather than a subclass. And in practice that would make the string immutable after construction, and I think that there may be room to dynamically change the way are globals are formatted based on context. Like a function putting a pointer to the module state into a local variable at the beginning of the function and then all further references within the function are relative to that pointer...